### ISSUE TITLE
Unresolved types examples

### ISSUE THREAD

**Issue by matklad** [intellij-rust/intellij-rust#1272]
Feel free to add isolated code snippets here, where the plugin fails to infer types correctly.

**Comment by vasily-kirichenko** [comment]
```rust
fn maybe() -> Option<i32> { Some(1) }

fn main() {
    if let Some(x) = maybe() {}
}
```
![image](https://cloud.githubusercontent.com/assets/873919/26521235/3ee507da-42ec-11e7-83ea-1ee9a1682c5f.png)

**Comment by farodin91** [comment]
#1376

**Comment by matklad** [comment]
The first example is fixed, and #1376 has a separate issue, so let's close this.
